sweep/tx_input_set.go

```diff
@@ -369,8 +369,14 @@ func (t *txInputSet) addPositiveYieldInputs(sweepableInputs []txInput) {
 }
 
 // tryAddWalletInputsIfNeeded retrieves utxos from the wallet and tries adding
-// as many as required to bring the tx output value above the given minimum.
+// as many as required to bring the tx output value above the given minimum if
+// this is a force sweep set.
 func (t *txInputSet) tryAddWalletInputsIfNeeded() error {
+	// Skip adding wallet input if this is not a force sweep.
+	if !t.force {
```

Comment:
If what we want here is to reduce unnecessary UTXO consumption by the sweeper due to stuff like anchors, then we can remove this section here that preemptively sends the anchor sweeps to the sweeper upon force close: https://github.com/lightningnetwork/lnd/blob/90effda090e752662500aa47eaed8b8619687e4e/contractcourt/channel_arbitrator.go#L1112-L1132

> We don't set those to Force because it makes no sense to sweep them if it's uneconomical

This isn't always the case. Normally yes, but if we _did_ make the decision to go to chain, then we _must_ sweeps those HTLCs as otherwise we won't be able to cancel back the HTLC on the incoming link. If mempools are congested, then this can lead to cascading force closes. 

This PR attempts to short circuit that and avoid going to chain at all. But if we _did_ decide to go to chain, then we _must_ force close.

Reply:
The current code doesn't set it to Force though, so the 2nd-level won't happen if it's not economical? As far as https://github.com/lightningnetwork/lnd/pull/7685 goes, I think it makes more sense to abandon the sweep and give up on the incoming HTLC because then you're not losing value_of_incoming_htlc + negative_outgoing_sweep and instead just lose value_of_incoming_htlc

Reply:
> This isn't always the case. Normally yes, but if we did make the decision to go to chain, then we must sweeps those HTLCs as otherwise 

Does this not only apply for Outgoing HTLCs, why should we care for HTLC-Success transactions if they are not worth sweeping economically ? There should be no incoming link at risk for the Success-Tx ?

Reply:
My bad, I was looking at a branch before https://github.com/lightningnetwork/lnd/pull/7966 was applied. So this should only apply to the incoming side. However, I still don't think it makes sense to sweep the outgoing HTLC at a loss when we should just take the loss and not sweep it if possible